gradle-typescript/src/main/java/com/gradlets/gradle/typescript/idea/CreateTsConfigTask.java

```diff
@@ -97,39 +80,24 @@ public final RegularFileProperty getTsConfig() {
 
     @TaskAction
     public final void generateTsConfig() throws IOException {
-        Configuration classpath = getCompileClasspath().get();
-        Set<File> dependenciesFiles = classpath.getIncoming().getArtifacts().getArtifacts().stream()
-                .filter(resolvedArtifact ->
-                        !(resolvedArtifact.getId().getComponentIdentifier() instanceof ProjectComponentIdentifier))
-                .map(ResolvedArtifactResult::getFile)
-                .collect(Collectors.toUnmodifiableSet());
-        Map<String, List<Path>> projectDeps = classpath.getIncoming().getResolutionResult().getAllComponents().stream()
-                .map(ComponentResult::getId)
-                .filter(ProjectComponentIdentifier.class::isInstance)
-                .map(ProjectComponentIdentifier.class::cast)
-                .map(ProjectComponentIdentifier::getProjectPath)
-                .map(projectPath -> getProject().findProject(projectPath))
-                .filter(Objects::nonNull)
-                .filter(projectDep -> projectDep.getExtensions().findByType(TypeScriptPluginExtension.class) != null
-                        && projectDep.getExtensions().findByType(PackageJsonExtension.class) != null)
-                .collect(Collectors.toMap(
-                        projectDep -> {
-                            PackageJsonExtension packageJsonExtension =
-                                    projectDep.getExtensions().findByType(PackageJsonExtension.class);
-                            return !packageJsonExtension.getScope().get().equals("")
-                                    ? "@" + packageJsonExtension.getScope().get() + "/" + projectDep.getName()
-                                    : projectDep.getName();
-                        },
-                        projectDep -> projectDep
-                                .getExtensions()
-                                .findByType(TypeScriptPluginExtension.class)
-                                .getSourceSets()
-                                .getByName("main")
-                                .getSource()
-                                .getSrcDirs()
-                                .stream()
-                                .map(File::toPath)
-                                .collect(Collectors.toUnmodifiableList())));
+        Set<File> dependenciesFiles = getCompileClasspath()
+                .filter(element -> !element.getAbsolutePath()
+                        .startsWith(
+                                getProject().getRootProject().getProjectDir().getAbsolutePath()))
+                .getFiles();
+
+        Map<String, List<Path>> projectDepsFiles = getCompileClasspath()
```

Comment:
filter after converting to a stream?

Reply:
I have put it before hoping that gradle does something smart here but can switch